### PR TITLE
[SPARK-28059][SQL][TEST] Port int4.sql

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

This PR is to port int4.sql from PostgreSQL regression tests. https://github.com/postgres/postgres/blob/REL_12_BETA1/src/test/regress/sql/int4.sql

The expected results can be found in the link: https://github.com/postgres/postgres/blob/REL_12_BETA1/src/test/regress/expected/int4.out

When porting the test cases, found two PostgreSQL specific features that do not exist in Spark SQL:
[SPARK-28023](https://issues.apache.org/jira/browse/SPARK-28023): Trim the string when cast string type to other types
[SPARK-28027](https://issues.apache.org/jira/browse/SPARK-28027): Add bitwise shift left/right operators

Also, found a bug:
[SPARK-28024](https://issues.apache.org/jira/browse/SPARK-28024): Incorrect value when out of range

Also, found four inconsistent behavior:
[SPARK-27923](https://issues.apache.org/jira/browse/SPARK-27923): Invalid input syntax for integer: "34.5" at PostgreSQL
[SPARK-28027](https://issues.apache.org/jira/browse/SPARK-28027) Our operator `!` and `!!` has different meanings
[SPARK-28028](https://issues.apache.org/jira/browse/SPARK-28028): Cast numeric to integral type need round
[SPARK-2659](https://issues.apache.org/jira/browse/SPARK-2659): HiveQL: Division operator should always perform fractional division, for example:
```sql
select 1/2;
```

## How was this patch tested?

N/A

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/inputs/pgSQL/int4.sql [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+-- bad input values -- should give errors
+-- INSERT INTO INT4_TBL(f1) VALUES ('1000000000000');
+-- INSERT INTO INT4_TBL(f1) VALUES ('asdf');
+-- INSERT INTO INT4_TBL(f1) VALUES ('     ');
+-- INSERT INTO INT4_TBL(f1) VALUES ('   asdf   ');
+-- INSERT INTO INT4_TBL(f1) VALUES ('- 1234');
+-- INSERT INTO INT4_TBL(f1) VALUES ('123       5');
+-- INSERT INTO INT4_TBL(f1) VALUES ('');
+
+select * from INT4_TBL;
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/pgSQL/int4.sql]: This does not exist in the original PostgreSQL test suite.

**Review by gatorsmile — COMMENTED** [review]
LGTM except one comment.

File: sql/core/src/test/resources/sql-tests/inputs/pgSQL/int4.sql (hunk 1 above)
Comment (wangyum) [reply sql/core/src/test/resources/sql-tests/inputs/pgSQL/int4.sql]: Removed it.

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**